opentelemetry-sdk/src/opentelemetry/sdk/trace/__init__.py

```diff
@@ -333,6 +334,13 @@ def __init__(
         self._current_span_slot = Context.register_slot(slot_name)
         self._active_span_processor = MultiSpanProcessor()
         self.sampler = sampler
+        self._atexit_hanlder = atexit.register(
+            self._active_span_processor.shutdown
+        )
+
+    def __del__(self):
```

Comment:
Why put this on `Tracer` instead of `SpanProcessor`?

Reply:
I think that having it in `Tracer` makes it more general and other possible `SpanProcessor` implementations would not have to take care of this.